pkg/sidecar/client_test.go

```diff
@@ -10,46 +15,140 @@ func getHost() string {
+						"operationState": "FINISHED"
+    				},
+					{
+						"type": "backup",
+						"id": "d3262073-8101-450f-9a11-c851760abd57",
+						"duration": "PT1M0.613S",
+						"start": "2019-06-11T03:37:15.593Z",
+						"stop": "2019-06-11T03:38:16.206Z",
+						"operationState": "RUNNING"
+    				}
+				]`))),
+			Status:     "200 OK",
+			StatusCode: 200,
+		},
+	}
+
+	// get operations
+
+	ops, err := client.GetOperations();
+	if err != nil {
+		t.Error(err)
+	}
+
+	backups, err := FilterOperations(*ops, backup)
+	assert.Assert(t, len(backups) == 1)
+
+	decommissions, err := FilterOperations(*ops, decommission)
+	assert.Assert(t, len(decommissions) == 1)
+
+	// get operation
+
+	client.testResponse = &resty.Response{
+		RawResponse: &http.Response{
+			Body: ioutil.NopCloser(bytes.NewReader([]byte(
+				`{
+					"type": "backup",
+					"id": "d3262073-8101-450f-9a11-c851760abd57",
+					"duration": "PT1M0.613S",
+					"start": "2019-06-11T03:37:15.593Z",
+					"stop": "2019-06-11T03:38:16.206Z",
+					"operationState": "RUNNING"
+				}`))),
+			Status:     "200 OK",
+			StatusCode: 200,
+		},
+	}
+
+	op, err := client.GetOperation("d3262073-8101-450f-9a11-c851760abd57")
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	if (*op)["operationState"] != "RUNNING" {
+		t.Errorf("testing backing should return RUNNING state")
+	}
+}
+
 func TestSidecarClient_GetStatus(t *testing.T) {
 
-	client := NewSidecarClient(getHost(), &ClientOptions{Secure: false, Port: 4567})
+	client := NewSidecarClient(getHost(), &DefaultSidecarClientOptions)
 
-	response, e := client.GetStatus()
+	status, e := client.Status()
 
 	if e != nil {
 		t.Errorf(e.Error())
-	} else if response == nil {
-		t.Errorf("GetStatus has not returned error but its response is not set.")
-	} else if response.OperationMode != "NORMAL" {
-		t.Errorf("GetStatus was successful but returned %s instead of %s", response.OperationMode, "NORMAL")
+	}
+
+	if status == nil {
+		t.Errorf("Status endpoint has not returned error but its status is not set.")
+	}
+
+	if status.OperationMode != OPERATION_MODE_NORMAL {
+		t.Fatalf("Expected NORMAL operation mode but received %v", status.OperationMode)
 	}
 }
 
 func TestClient_DecommissionNode(t *testing.T) {
 
-	client := NewSidecarClient(getHost(), &ClientOptions{Secure: false, Port: 4567})
+	client := NewSidecarClient(getHost(), &DefaultSidecarClientOptions)
 
 	// first decommissioning
 
-	response, err := client.DecommissionNode()
-
-	if err != nil {
+	if response, err := client.Decommission(); err != nil {
 		t.Errorf(err.Error())
-	}
-
-	if response == nil {
-		t.Errorf("Decommissioning was successful but no DecommissionResponse was returned.")
+	} else if response == nil {
+		t.Errorf("there is not any error from Decommission endpoint but response is nil")
+	} else if getOpResponse, err := client.GetOperation(*response); err != nil {
+		t.Errorf(err.Error())
+	} else {
+		assert.Assert(t, (*getOpResponse)["operationState"] == "RUNNING")
```

Comment:
Does this "RUNNING" reflect the operations' state? Meaning, the `decomissioning` is in progress?